src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,114 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
 
 export const PeoplePage = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [erroeMessage, setErrorMessage] = useState('');
+
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => setErrorMessage('Something went wrong'))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const order = searchParams.get('order') || '';
+  const sex = searchParams.get('sex');
+  const century = searchParams.getAll('centuries');
+  const query = searchParams.get('query');
+  const sort = searchParams.get('sort');
+
+  const queryIncludes = (str = '', person: Person) => {
+    return (person.name.toLowerCase().includes(str)
+      || person.fatherName?.toLowerCase().includes(str)
+      || person.motherName?.toLowerCase().includes(str)
+    );
+  };
+
+  const bornInCentury = (centuries: string[], person: Person) => {
+    return (!centuries.length
+      || centuries.includes(Math.ceil(person.born / 100).toString())
+    );
+  };
+
+  const preparedPeople = useMemo(() => {
+    const filtredAndSortedPeople = people.filter((person) => {
+      const lowerQuery = query?.toLowerCase();
+
+      return (
+        (sex ? person.sex === sex : person)
+        && queryIncludes(lowerQuery, person)
+        && bornInCentury(century, person)
+      );
+    })
+      .sort((person1, person2) => {
```

Comment:
sort is mutable method, so make a copy of people array

Reply:
done